src/core/crypto/signature.ts

```diff
@@ -0,0 +1,53 @@
+import { Serializable } from "../../bcs";
+import { AuthenticationKeyScheme, SigningSchemeInput } from "../../types";
+import { Hex } from "../hex";
+
+/**
+ * An abstract representation of a crypto signature,
+ * associated to a specific signature scheme e.g. Ed25519 or Secp256k1
+ *
+ * This is the product of signing a message directly from a PrivateKey
+ * and can be verified against a CryptoPublicKey.
+ */
+export abstract class CryptoSignature extends Serializable {
+  abstract readonly signatureScheme: SigningSchemeInput;
+
+  /**
+   * Get the raw signature bytes
+   */
+  abstract toUint8Array(): Uint8Array;
+
+  /**
+   * Get the signature as a hex string with a 0x prefix e.g. 0x123456...
+   */
+  toString(): string {
+    const bytes = this.toUint8Array();
+    return Hex.fromHexInput(bytes).toString();
+  }
+}
+
+/**
+ * An abstract representation of an account signature,
+ * associated to a specific authentication scheme e.g. Ed25519 or SingleKey
+ *
+ * This is the product of signing a message through an account,
+ * and can be verified against an AccountPublicKey.
+ */
+export abstract class AccountSignature extends Serializable {
+  abstract readonly authenticationScheme: AuthenticationKeyScheme;
+
+  /**
+   * Get the raw signature bytes
+   * TODO (gabi): do we need this?
```

Comment:
I'm keeping the function around to avoid breaking changes,
but also I don't want to make a decision whether to remove it or not in this PR to avoid slowing the process.

Reply:
There's already `bcsToBytes()` for that purpose.
Transmitting the raw bytes of the signature without any indication of which signature scheme it's based off could also be confusing.

Reply:
Just checked the Rust code. `AnySignature` does not have a "to_bytes" function, although `AnyPublicKey" does.
I can just keep it around and consistent with `AnyPublicKey`, for now